### ISSUE TITLE
Board views are not being rendered: (skipping missing collection view for block)

### ISSUE THREAD

**Issue by pbteja1998** [NotionX/react-notion-x#104]
Public Notion Page Link: https://www.notion.so/b966297c5e874435837c9c531cd0c25f

I just discovered that sometimes board views are not being rendered with `react-notion-x`. I am not sure if I have to enable something for it to work. 

FYI:
- I have already added `Collection` and `CollectionRow` components to `NotionRenderer`. 
- This seems to be happening only for Board views. Gallery and Table views are working properly.
- I am getting the error message `skipping missing collection view for block`

It would be really helpful if someone can point me to what is happening here.

Thanks.

**Comment by schickling** [comment]
Getting the same error for Table views. (Worked previously.)

**Comment by schickling** [comment]
I've investigated things a little bit and it seems that the `collection` property is no longer fetched properly: 
![CleanShot 2021-07-09 at 11 07 11](https://user-images.githubusercontent.com/1567498/125053859-dc965f00-e0a5-11eb-9052-950eb903b957.png)

When comparing results with the DevTools results on an opened Notion web page, I've noticed that they've seemed to changed the API (which was kind of to expected given it's an unofficial API). Instead of the `loadPageChunk` API (which is used by the package) the Notion web app seems to use a `loadCachedPageChunk` API.

![image](https://user-images.githubusercontent.com/1567498/125054198-3d259c00-e0a6-11eb-94ca-80f3355776e8.png)

**Comment by schickling** [comment]
@normdoow as you seem to be mostly maintaining this lib right now - any chance you could take a look into this? 👀

**Comment by pbteja1998** [comment]
> @normdoow as you seem to be mostly maintaining this lib right now - any chance you could take a look into this? 👀

Yes, it would be great if it can be looked into.

**Comment by normdoow** [comment]
@pbteja1998 your test page is private. Can you make it public again?

**Comment by pbteja1998** [comment]
> @pbteja1998 your test page is private. Can you make it public again?

Can you try now?

**Comment by normdoow** [comment]
@pbteja1998 my guess is that it has something to do with your boards being linked databases. The database still needs to be a database that is a child of the main notion page. If its linking to a database outside of that, it won't show.

**Comment by pbteja1998** [comment]
Hey @normdoow, thank you for looking into it. In the page I shared, one of the boards is an inline database, it's not a linked database. Even that is not being rendered.

**Comment by pbteja1998** [comment]
Check the database that is at the end. It's named `Test`.

**Comment by pbteja1998** [comment]
Also, there is a `table view` which is using linked database, that is being rendered properly. But the `board view` of the same linked database is not being rendered – which makes me think that the problem is with rendering boards, rather than the database being a linked database. 

What do you think?

**Comment by normdoow** [comment]
> Also, in the same demo page, there is a `table view` which is using a linked database, that is being rendered properly. But the `board view` of the same linked database is not being rendered – which makes me think that the problem is with rendering boards, rather than the database being a linked database.
> 
> What do you think?

I'm seeing the same. Though I know inline boards are working elsewhere. Haven't been able to dive deep into this yet.

**Comment by pbteja1998** [comment]
Hey @normdoow, may be it's happening only for the newly created boards. Like @schickling has suggested, something might have changed in their API. Anyways, thanks for looking into it. Please let me know when you find some time to dig deeper into it. Meanwhile, I will also try to figure out a solution to this.

**Comment by normdoow** [comment]
@pbteja1998 I am starting to see this be an issue. It will be one of the next things I work on. 👌

**Comment by normdoow** [comment]
@pbteja1998 and @schickling, could you test this to make sure it works? https://github.com/NotionX/react-notion-x/pull/107

**Comment by schickling** [comment]
Thanks a lot. Is there a pre-release that can be installed via npm?

**Comment by pbteja1998** [comment]
Hey @normdoow, thanks a lot for working on this. Is there any way to test this? or should we just build the repo locally and see?

**Comment by normdoow** [comment]
@schickling, @pbteja1998. Yeah, so pulling the branch from the PR and then testing locally is really the only way to test this right now. Nothing on NPM currently. Maybe it could make sense for us to do something like that if we get enough people collaborating on this.

**Comment by schickling** [comment]
Got it @normdoow. So I've ended up spending some time publishing the branch from #107 to the following NPM packages:

https://www.npmjs.com/package/@schickling-tmp/react-notion-x
https://www.npmjs.com/package/@schickling-tmp/notion-utils
https://www.npmjs.com/package/@schickling-tmp/notion-types
https://www.npmjs.com/package/@schickling-tmp/notion-client

(I invite others to do some testing based on my published packages.)

---

I've then tried using the new version in my application - which however unfortunately didn't fix the problem. Still seeing the same issue.

Side note: @normdoow given you're already using lerna it might be a lot simpler to publish some pre-releases for the WIP branch for some testing/debugging here than having to go through what I've done above as it involves quite a lot of manual work

![image](https://user-images.githubusercontent.com/1567498/126875039-5a81495b-4ca5-43c8-a6db-87bb84aa848c.png)

**Comment by normdoow** [comment]
@schickling, thanks for this. Yeah it may make since to do some pre-release packages that are automatic.

I have been testing this as well. Currently I'm just getting the headers but not the cards.
![image](https://user-images.githubusercontent.com/21371266/126875357-232d1138-025f-4052-9afb-eb871883047e.png)

**Comment by normdoow** [comment]
@schickling @pbteja1998 This should be fixed now on the PR. had to update group_by in the query as well.

Make sure you have the Collection and CollectionRow components set in NotionRenderer when testing as well.
![image](https://user-images.githubusercontent.com/21371266/126878070-14ae7e03-5585-4961-a756-edd33460647a.png)

**Comment by schickling** [comment]
I've pulled, re-built and tried again but unfortunately to no avail. Still seeing `skipping missing collection view for block ...`

**Comment by normdoow** [comment]
> I've pulled, re-built and tried again but unfortunately to no avail. Still seeing `skipping missing collection view for block ...`

and you've set the compontents in NotionRenderer?
![image](https://user-images.githubusercontent.com/21371266/126880273-59427881-fdc1-43d4-a072-c1b53e823f38.png)

**Comment by schickling** [comment]
Yes should be: 
![CleanShot 2021-07-24 at 22 30 49](https://user-images.githubusercontent.com/1567498/126880431-afa776d1-3792-48c9-a0b9-c921a115942e.png)

**Comment by normdoow** [comment]
Ok. thats good. Can you share a public notion page that you are testing on?
Looks like one of these objects is falsey for you. Which I'm not seeing on my usecase.
![image](https://user-images.githubusercontent.com/21371266/126880507-a77a64c1-d2fe-400a-adf6-48e2d5916af1.png)

**Comment by schickling** [comment]
I've taken a closer look and it turns out my problem was most likely unrelated to the thread above.

TLDR: I'm SSRing **private pages** which requires the following parameter for the `NotionAPI` client. The problem was that the `token_v2` value expired earlier than expected leading to the error message I've posted above.

```ts
import { NotionAPI } from 'notion-client'

export const notion = new NotionAPI({
  // extract from Notion cookies
  authToken: process.env['NOTION_AUTH_TOKEN'],
  activeUser: process.env['NOTION_USER_ID'],
})

```

Thanks a lot for looking into this regardless @normdoow 💯

**Comment by normdoow** [comment]
@pbteja1998 this is now fixed in v4.7.10

**Comment by pbteja1998** [comment]
Hey @normdoow completely missed this. Thanks a lot for resolving this. I just checked in my user's site, the board is now being shown properly.

**Comment by datduyng** [comment]
FYI, for those that is still seeing this problem  in react-notion-x: ^6.5.0. Seems like we currently don't support table view block in which it link to an existing table

i.e

I have this table below. I want to reference it on a Notion page but It give me this error


<img width="1038" alt="CleanShot 2022-04-02 at 13 31 23@2x" src="https://user-images.githubusercontent.com/35666615/161400212-6ffaaa49-cc87-4124-ba55-ccca75e16513.png">


<img width="814" alt="CleanShot 2022-04-02 at 13 32 05@2x" src="https://user-images.githubusercontent.com/35666615/161400241-c4ac8fa1-4e53-4927-90f1-9f4bf07c4ee8.png">


<img width="650" alt="CleanShot 2022-04-02 at 13 32 47@2x" src="https://user-images.githubusercontent.com/35666615/161400251-edb0282e-85f2-4364-90b8-d3fef1813645.png">

**Comment by transitive-bullshit** [comment]
@datduyng thanks for the info. Since this is a different bug from the one in this thread, let's track it in #260
